### PR TITLE
migrate transfer process screen to compose screen #2583

### LINKED ISSUE #2583: refactor: migrate TransferProcessFragment to compose screen
**Summary:**

migrate TransferProcessFragment to compose screen

### PR DESCRIPTION
**Pull request by Saifuddin53**
Fixes #2583 

Please Add Screenshots If there are any UI changes.

Before : 

before 

After :

after

Videos: 

https://github.com/openMF/mifos-mobile/assets/112172255/32038de8-df5f-4e34-a992-59746c845eb5

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+
+
+    /**
+     * Closes the transfer fragment
+     */
+    private fun closeClicked() {
+        activity?.supportFragmentManager?.popBackStack()
+        activity?.supportFragmentManager?.popBackStack()
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt]: typo: two same lines

File: app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                MifosMobileTheme {
+                    TransferProcessScreen(
+                        transfer = {
+                            viewModel.makeTransfer(payload)
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt]: make this function of viewModel in the compose screen `TransferProcessScreen`,  as we have the instance of viewModel there

File: app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+/**
+ * Created by dilpreet on 1/7/17.
+ */
+@AndroidEntryPoint
+class TransferProcessComposeFragment : BaseFragment() {
+
+    private val viewModel: TransferProcessViewModel by viewModels()
+
+    private lateinit var payload: TransferPayload
+    private var transferType: TransferType? = null
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt]: move this to viewModel as well

File: app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    private val _transferUiState = MutableStateFlow<TransferUiState>(TransferUiState.Initial)
+    val transferUiState: StateFlow<TransferUiState> get() = _transferUiState
+
+    private var _transferPayload: MutableStateFlow<TransferPayload?> = MutableStateFlow(null)
+    val transferPayload: StateFlow<TransferPayload?> get() = _transferPayload
+
+    private var _transferType: MutableStateFlow<TransferType?> = MutableStateFlow(null)
+    val transferType: StateFlow<TransferType?> get() = _transferType
+    fun makeTransfer(payload: TransferPayload) {
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessViewModel.kt]: use proper spacing between lines, and beautify text in Android studio before any commit

**Review by AvneetSingh2001 — CHANGES_REQUESTED** [review]

**Comment by AvneetSingh2001** [comment]
@Saifuddin53  requested some change, please update the code accordingly. Also can you please add screen recording as well

**Comment by AvneetSingh2001** [comment]
@Saifuddin53 , can you check why are we getting HTTP 400 response for this?

**Comment by Saifuddin53** [comment]
I think it's a server side error

**Comment by AvneetSingh2001** [comment]
can you paste the logs about the error here

**Comment by Saifuddin53** [comment]
Screenshot 2024-05-15 084635

**Comment by Saifuddin53** [comment]
The same error is coming in our view-based screen. It would be solved when we migrate our previous screen which is SavingsMakeTransferFragment as it is passing **null** for the payto and payfrom fields

**Comment by Saifuddin53** [comment]
https://github.com/openMF/mifos-mobile/assets/112172255/5df044c6-40c2-4fa7-93b6-b8301b0f8dc5

File: app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+
+
+    /**
+     * Closes the transfer fragment
+     */
+    private fun closeClicked() {
+        activity?.supportFragmentManager?.popBackStack()
+        activity?.supportFragmentManager?.popBackStack()
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt]: same lines still

File: app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessScreen.kt [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+
+                is TransferUiState.TransferSuccess -> {
+                    Toast.makeText(context, R.string.transferred_successfully, Toast.LENGTH_SHORT).show()
+                }
+
+                is TransferUiState.Error -> {
+                    ErrorComponent(errorToast = uiState.errorMessage.message)
+                }
+
+                is TransferUiState.Initial -> {}
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessScreen.kt]: use Unit, 
                is TransferUiState.Initial ->  Unit

**Comment by Saifuddin53** [comment]
Screenshot 2024-05-15 214738
@AvneetSingh2001 here the showSavingsAccountTemplate function is not able to fetch the accounts from the AccountOptions list

File: app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                MifosMobileTheme {
+                    TransferProcessScreen(
+                        cancel = {
+                                closeClicked()
```
Comment (itsPronay) [inline app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt]: I think replacing this with `activity?.supportFragmentManager?.popBackStack()` is a better option.

File: app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessScreen.kt [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+                            transfer()
+                        }
+                    ) {
+                        Text(text = stringResource(id = R.string.transfer))
+                    }
+                }
+            }
+        }
+    }
+}
```
Comment (itsPronay) [inline app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessScreen.kt:221]: generate previews

**Comment by Saifuddin53** [comment]
@AvneetSingh2001 can you please review it again

**Comment by itsPronay** [comment]
@Saifuddin53 I see you have multiple commits. Please squash them

File: app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+                        },
+                    )
+                }
+            }
+        }
+    }
+
+
```
Comment (itsPronay) [inline app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt]: Hide toolbar on resume

**Review by itsPronay — CHANGES_REQUESTED** [review]
image

fix this one as well. 
 There's an issue with the layout when the device is rotated. Only a portion of the screen remains visible, and scrolling is disabled because you're using a Column.

File: app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt (hunk 9 above)
Comment (itsPronay) [reply app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessComposeFragment.kt]: and for toolbar use scaffold

**Comment by itsPronay** [comment]
also please reformat your code, I believe it's ctrl + alt + L

File: app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessScreen.kt [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.End
+            ) {
+                Row(
+                    horizontalArrangement = Arrangement.spacedBy(30.dp),
+                ) {
+                    Button(
+                        onClick = {
+                                  cancel()
```
Comment (itsPronay) [inline app/src/main/java/org/mifos/mobile/ui/transfer_process/TransferProcessScreen.kt]: image
you didn't implement this snackbar.. Original (XML) one has it... add this as well

**Pull request merged**